### PR TITLE
chore: add error reporting of caught exceptions

### LINKED ISSUE #4174: Log every possible error and warning and improve error handling
The error reporting story of Open Event ecosystem is a sad state. In both server and frontend, generic errors are just caught and shown to the user who are presented with `HTTP POST 409 https://akldjflksdfl...` or `An unexpected error occured`. No one is none the wiser. Devs cannot tell what was wrong and what caused the system to error out because there are no logs on both frontend or server. Order was failed, but no logs. Discount code usage exceeded, but no logs. Tickets for event exhausted, but no logs

It is as if someone deliberately told devs to not use logging and error reporting in both frontend and server. Nothing gets logged, even on critical errors, and only unhandled errors are reported, because it is not possible to suppress them, or else I think they would be suppressed as well.

Goal is to:
- [ ] Log every possible error and even warning
- [ ] Reject pull requests if there is a generic catch block without logging
- [ ] Create a generic method which logs the error, parses it and shown proper error to user, at least what was sent from server, not just `HTTP POST 409 https://akldjflksdfl...` or `An unexpected error occured`
- [ ] Replace that generic error handler with specific user-friendly messages with all possible courses of action

### PR DESCRIPTION
**Pull request by Dishebh**
Fixes part of issue #4174

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Really?

File: app/components/account/application-section.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ export default Component.extend({
           });
         }
       } catch (error) {
+        console.error("Error during authentication", error);
```
Comment (niranjan94) [inline app/components/account/application-section.js]: Codacy Issue found: [Strings must use singlequote.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5212015)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5212015&bid=17090611)

**Comment by Dishebh** [comment]
> Really?

Updated description.

**Comment by iamareebjamal** [comment]
Do you think this fixes the issue?

**Comment by Dishebh** [comment]
> Do you think this fixes the issue?

A part of the issue?

**Comment by iamareebjamal** [comment]
Not even 1%. I cannot merge this PR. Please make meaningful PRs, not one line changes

**Comment by Dishebh** [comment]
Shall I close this, then?

**Comment by iamareebjamal** [comment]
If you don't want to contrinue on this, yes

**Pull request closed without merging**